### ISSUE TITLE
weighting.from_simprod fails for old datasets (Trac #2100)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2092]
weighting.from_simprod fails fot dataset 6475

```text
from icecube.weighting import weighting
weighting.from_simprod(6475)

ValueError                                Traceback (most recent call last)
<ipython-input-4-9e953071698d> in <module>()
----> 1 weighting.from_simprod(6475)

/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/combo/trunk/lib/icecube/weighting/weighting.py in from_simprod(dataset_id, use_muongun)
    638                         nugen_kwargs['CylinderHeight'] = steering['NUGEN::cylinder_length']
    639                         nugen_kwargs['CylinderRadius'] = steering['NUGEN::cylinder_radius']
--> 640                 if get_metaproject(dataset_id, 'nugen', 0)[1:] >= (4,1,6):
    641                         nugen_kwargs['InjectionMode'] = 'Cylinder'
    642                 generator = NeutrinoGenerator(NEvents=steering['nevents'],

/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/combo/trunk/lib/icecube/weighting/weighting.py in get_metaproject(dataset_id, tray_name, tray_index)
    623                         cursor.execute("SELECT metaproject.name, metaproject.major_version, metaproject.minor_version, metaproject.patch_version FROM tray JOIN metaproject_pivot ON tray.tray_index=metaproject_pivot.tray_index AND tray.dataset_id=metaproject_pivot.dataset_id JOIN metaproject ON metaproject_pivot.metaproject_id=metaproject.metaproject_id WHERE tray.dataset_id=%s AND tray.tray_index=%s", (dataset_id, tray_index))
    624                         row = cursor.fetchone()
--> 625                 return tuple([row[0]] + map(int, row[1:]))
    626
    627         if category == 'neutrino-generator':

ValueError: invalid literal for int() with base 10: '09-RC'

```

Migrated from https://code.icecube.wisc.edu/ticket/2100
```json
{
    "status": "closed", 
    "changetime": "2017-10-23T15:32:25", 
    "description": "weighting.from_simprod fails fot dataset 6475\n\n\n{{{\nfrom icecube.weighting import weighting\nweighting.from_simprod(6475)\n\nValueError                                Traceback (most recent call last)\n<ipython-input-4-9e953071698d> in <module>()\n----> 1 weighting.from_simprod(6475)\n\n/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/combo/trunk/lib/icecube/weighting/weighting.py in from_simprod(dataset_id, use_muongun)\n    638                         nugen_kwargs['CylinderHeight'] = steering['NUGEN::cylinder_length']\n    639                         nugen_kwargs['CylinderRadius'] = steering['NUGEN::cylinder_radius']\n--> 640                 if get_metaproject(dataset_id, 'nugen', 0)[1:] >= (4,1,6):\n    641                         nugen_kwargs['InjectionMode'] = 'Cylinder'\n    642                 generator = NeutrinoGenerator(NEvents=steering['nevents'],\n\n/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/combo/trunk/lib/icecube/weighting/weighting.py in get_metaproject(dataset_id, tray_name, tray_index)\n    623                         cursor.execute(\"SELECT metaproject.name, metaproject.major_version, metaproject.minor_version, metaproject.patch_version FROM tray JOIN metaproject_pivot ON tray.tray_index=metaproject_pivot.tray_index AND tray.dataset_id=metaproject_pivot.dataset_id JOIN metaproject ON metaproject_pivot.metaproject_id=metaproject.metaproject_id WHERE tray.dataset_id=%s AND tray.tray_index=%s\", (dataset_id, tray_index))\n    624                         row = cursor.fetchone()\n--> 625                 return tuple([row[0]] + map(int, row[1:]))\n    626\n    627         if category == 'neutrino-generator':\n\nValueError: invalid literal for int() with base 10: '09-RC'\n\n}}}\n\n\n", 
    "reporter": "chaack", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1508772745265055", 
    "component": "combo reconstruction", 
    "summary": "weighting.from_simprod fails for old datasets", 
    "priority": "normal", 
    "keywords": "weighting", 
    "time": "2017-10-16T13:03:18", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20171016T13:33:41: chaack commented:

> also weighting.py uses tabs instead of spaces :'(

**Comment by icecube-trac** [comment]
Trac update at 20171016T13:55:18: 
* jvansanten changed owner from "" to "jvansanten"
* jvansanten commented:

> In changeset 158683:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="158683"
> Parse prerelease version out of metaproject patch field if it's
> there.Fixes #2092.
> ```


* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20171023T15:32:25: nega commented:

> In changeset 158966:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="158966"
> stop module from blowing up when there is no prerelease version in the patch field
> 
> see 158683/IceCube
> fixes #2092
> ```
